File: client/src/api/samba.ts
import { apiClient } from '../lib/api';

export interface SambaConnection {
  pid: string;
  username: string;
  machine: string;
}

export interface SambaStatus {
  is_running: boolean;
  version: string | null;
  active_connections: SambaConnection[];
  smb_users_count: number;
}

export interface SambaUserStatus {
  user_id: number;
  username: string;
  role: string;
  smb_enabled: boolean;
  is_active: boolean;
}

export interface SambaUsersResponse {
  users: SambaUserStatus[];
}

export interface OsConnectionInfo {
  os: string;
  label: string;
  command: string;
  notes: string | null;
}

export interface SambaConnectionInfo {
  is_running: boolean;
  share_name: string;
  smb_path: string;
  username: string;
  instructions: OsConnectionInfo[];
}

export async function getSambaStatus(): Promise<SambaStatus> {
  const { data } = await apiClient.get<SambaStatus>('/api/samba/status');
  return data;
}

export async function getSambaUsers(): Promise<SambaUsersResponse> {
  const { data } = await apiClient.get<SambaUsersResponse>('/api/samba/users');
  return data;
}

export async function toggleSmbUser(
  userId: number,
  enabled: boolean,
  password?: string,
): Promise<{ user_id: number; username: string; smb_enabled: boolean }> {
  const { data } = await apiClient.post(`/api/samba/users/${userId}/toggle`, {
    enabled,
    password: password || undefined,
  });
  return data;
}

export async function getSambaConnectionInfo(): Promise<SambaConnectionInfo> {
  const { data } = await apiClient.get<SambaConnectionInfo>('/api/samba/connection-info');
  return data;
}

File: backend/app/schemas/samba.py
"""Pydantic models for Samba (SMB/CIFS) API."""

from typing import Optional

from pydantic import BaseModel

from app.schemas.webdav import OsConnectionInfo


class SambaConnection(BaseModel):
    """A single active SMB connection."""
    pid: str
    username: str
    machine: str


class SambaStatusResponse(BaseModel):
    """Samba server status (admin only)."""
    is_running: bool
    version: Optional[str] = None
    active_connections: list[SambaConnection] = []
    smb_users_count: int = 0


class SambaUserStatus(BaseModel):
    """SMB status for a single user."""
    user_id: int
    username: str
    role: str
    smb_enabled: bool
    is_active: bool


class SambaUsersResponse(BaseModel):
    """List of users with their SMB status."""
    users: list[SambaUserStatus]


class SambaUserToggleRequest(BaseModel):
    """Request to toggle SMB access for a user."""
    enabled: bool
    password: Optional[str] = None


class SambaConnectionInfo(BaseModel):
    """SMB connection info with per-OS mount instructions."""
    is_running: bool
    share_name: str
    smb_path: str
    username: str
    instructions: list[OsConnectionInfo]
